worker/uniter/uniter.go

```diff
@@ -275,6 +289,9 @@ func newUniter(uniterParams *UniterParams) func() (worker.Worker, error) {
 
 func (u *Uniter) loop(unitTag names.UnitTag) (err error) {
 	defer func() {
+		if r := recover(); r != nil {
+			err = errors.Errorf("panic: %v\n%v", r, string(debug.Stack()))
+		}
```

Comment:
I though we had this in the catacomb to do this generically for all workers

Reply:
My mistake was trying to catch something tricky.